protected/humhub/modules/tour/Events.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * @link https://www.humhub.org/
+ * @copyright Copyright (c) HumHub GmbH & Co. KG
+ * @license https://www.humhub.com/licences
+ */
+
+namespace humhub\modules\tour;
+
+use humhub\modules\dashboard\widgets\Sidebar;
+use humhub\modules\tour\widgets\Dashboard;
+use humhub\modules\user\models\User;
+use Yii;
+
+/**
+ * Events provides callbacks for all defined module events.
+ *
+ * @since 1.15
+ */
+class Events
+{
+    const AUTO_START = 'autoStartTour';
+
+    public static function onDashboardSidebarInit($event)
+    {
+        if (Yii::$app->user->isGuest) {
+            return;
+        }
+
+        if (self::getModule()->settings->user()->get(self::AUTO_START)) {
+            self::runAutoStartWelcomeTour();
+        } elseif(self::shouldDisplayDashboardWidget()) {
+            /* @var Sidebar $sidebar */
+            $sidebar = $event->sender;
+            $sidebar->addWidget(Dashboard::class, [], ['sortOrder' => 100]);
+        }
+    }
+
+    public static function onUserBeforeLogin($event)
+    {
+        if ($event->identity instanceof User && self::shouldStartWelcomeTour($event->identity)) {
+            self::getModule()->settings->user($event->identity)->set(self::AUTO_START, true);
+        }
+    }
+
+    private static function getModule(): Module
+    {
+        return Yii::$app->getModule('tour');
+    }
+
+    private static function shouldDisplayDashboardWidget(?User $user = null): bool
+    {
+        $settings = self::getModule()->settings;
+        return $settings->get('enable') == 1 &&
+            $settings->user($user)->get('hideTourPanel') != 1;
+    }
+
+    private static function shouldStartWelcomeTour(?User $user = null): bool
+    {
+        return $user->last_login === null && // Force auto start only for new created user who is logged in first time after registration
+            self::shouldDisplayDashboardWidget($user) && // Start it only when the dashboard sidebar widget is visible for the user
+            !self::getModule()->showWelcomeWindow($user); // No need auto start because it will be done by dashboard widget
+    }
+
+    private static function runAutoStartWelcomeTour(): void
+    {
+        self::getModule()->settings->user()->delete(self::AUTO_START);
+        Yii::$app->response->redirect(['/dashboard/dashboard', 'tour' => true]);
```

Comment:
@yurabakhtin Do we really need this extra redirect here?

Reply:
@luke- I find the GET flag `tour` is used to start it - https://github.com/humhub/humhub/blob/fix/6678-tour-auto-start-test-issue/protected/humhub/modules/tour/widgets/Tour.php#L26-L29:

```php
        // Active tour flag not set
        if (!isset($_GET['tour'])) {
            return;
        }
```

If you look at the Tour Dashboard widget:

![tour-widget](https://github.com/humhub/humhub/assets/6995524/f6beb180-42d4-4d67-864a-2b07229d19a4)

there all links have URL with param `?tour=1`.

Do you want to find a solution to keep all these 4 links without the flag `tour=1`?
Or do you want to keep the redirect e.g. on the event `onUserAfterLogin` instead of current `onDashboardSidebarInit`?

Reply:
I would just like to get rid of the unnecessary redirect. 

1. login (first)
2. dashboard
3. dashboard?tour=1

--- 

Why not something like
https://github.com/humhub/humhub/blob/fix/6678-tour-auto-start-test-issue/protected/humhub/modules/tour/widgets/Tour.php#L26-L29

```
        // Active tour flag not set
        if (!isset($_GET['tour']) || !$this->doAutostart()) {
            return;
        }
```

Reply:
@luke- Ok, I got it, thanks for the explanation. Done in commit https://github.com/humhub/humhub/pull/6685/commits/bef0799a4ae410c7b2b95a9bb2cbcabe6adf1f7a.